### ISSUE TITLE
testFilters is not using the correct ColumnStatistics

### ISSUE THREAD

**Issue by yingsu00** [facebookincubator/velox#11116]
### Bug description

SplitReader uses file level ColumnStatistics to test the filters to eliminate unqualified splits. This uses HiveConnectorUtil::testFilters(). This function get the ColumnStatistics using 
```
const auto columnStats = reader->columnStatistics(typeWithId->id());
```
reader->columnStatistics() needs to take the column Id, not the id in typeWithId. The id in typeWithId is the traversal sequence number in the type tree, not the index of the columns.

### System information

N/A

**Comment by Yuhta** [comment]
It's a bug in the comment, this should be the traversal ID so that subfield can also be filtered in the future.

**Comment by yingsu00** [comment]
> It's a bug in the comment, this should be the traversal ID so that subfield can also be filtered in the future.

Does DWRF or ORC have any statistics at the subfield level? Parquet only has column level statistics. I'm not aware that ORC, DWRF or Alpha has that level of statistics.
Also if you think the comment was wrong, then the `columnStats` and  `reader->columnStatistics()` should be renamed to avoid confusion.

**Comment by Yuhta** [comment]
Yes DWRF can have stats on subfields, we can rename these names to something like `fieldStatistics`
